packages/bsky/src/api/app/bsky/feed/getFeed.ts

```diff
@@ -243,14 +245,20 @@ const skeletonFromFeedGen = async (
   }
 
   const { feed: feedSkele, ...skele } = skeleton
-  const feedItems = feedSkele.map((item) => ({
-    post: { uri: item.post },
-    repost:
-      typeof item.reason?.repost === 'string'
-        ? { uri: item.reason.repost }
-        : undefined,
-    feedContext: item.feedContext,
-  }))
+  // Limit again in case the feedgen does not respect the limit.
+  // Leave some margin in case filtering removes more items.
+  const limitMargin = Math.ceil(params.limit * 0.3)
```

Comment:
Even though it's possible that some items will be filtered out, we expect feed gens to respect the `limit` passed to them regardless.  So perhaps it will be clearer to not provide a margin, rather than incentivize feed gens to serve oversized pages.  The built-in `app.bsky.feed.getTimeline` endpoint has the same limitation, for what it's worth.